File: DSA/Searching/Linear_Search.c
#include <stdio.h>

int linearsearch (int arr[], int n, int key) {
    for(int i=0; i<n; i++) {
        if(arr[i] == key) {
            return i;
        }
    }
    return -1;
}

int main () {
    int arr[] = {6, 10, 20, 30, 55, 75, 190};
    int n = sizeof(arr) / sizeof(arr[0]);
    int key;
    printf("Enter the key to search: ");
    scanf("%d", &key);

    int index = linearsearch(arr, n, key);
    if(index != -1) {
        printf("Element found at index: %d\n", index);
    } else {
        printf("Element not found.\n");
    }
}

File: DSA/Searching/Java/LinearSearch.java
import java.util.Scanner;

public class LinearSearch {
    static int linearSearch(int[] arr, int size, int key) {
        for(int i=0; i<size; i++){
            if(arr[i] == key) {
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        int[] arr = {10, 6, 44, 55, 77, 24};
        int size = arr.length;
        int key;
        key = input.nextInt();

        int index = linearSearch(arr, size, key);

        if(index != -1) {
            System.out.println("Element found at " + index);
        } else {
            System.out.println("Element not found");
        }
    }
}
